amptorch/trainer.py

```diff
@@ -82,59 +94,99 @@ def get_unique_elements(self, training_images):
+                )
+            elif self.cache == "no":
+                self.train_dataset = AtomsLMDBDataset(
+                    self.config["dataset"]["lmdb_path"],
+                )
+            else:
+                raise NotImplementedError
+            self.elements = self.train_dataset.elements
+            descriptor_setup = self.train_dataset.descriptor_setup
+        else:
+            training_images = self.config["dataset"]["raw_data"]
+            # TODO: Scalability when dataset to large to fit into memory
+            if isinstance(training_images, str):
+                training_images = ase.io.read(training_images, ":")
+            del self.config["dataset"]["raw_data"]
+
+            self.elements = self.config["dataset"].get(
+                "elements", self.get_unique_elements(training_images)
+            )
 
-        self.forcetraining = self.config["model"].get("get_forces", True)
-        self.fp_scheme = self.config["dataset"].get("fp_scheme", "gaussian").lower()
-        self.fp_params = self.config["dataset"]["fp_params"]
-        self.save_fps = self.config["dataset"].get("save_fps", True)
-        self.cutoff_params = self.config["dataset"].get(
-            "cutoff_params", {"cutoff_func": "Cosine"}
-        )
-        descriptor_setup = (
-            self.fp_scheme,
-            self.fp_params,
-            self.cutoff_params,
-            self.elements,
-        )
-        self.train_dataset = AtomsDataset(
-            images=training_images,
-            descriptor_setup=descriptor_setup,
-            forcetraining=self.forcetraining,
-            save_fps=self.config["dataset"].get("save_fps", True),
-            scaling=self.config["dataset"].get(
-                "scaling", {"type": "normalize", "range": (0, 1)}
-            ),
-        )
+            self.forcetraining = self.config["model"].get("get_forces", True)
+            # self.pca_reduce = self.config["dataset"].get("pca_reduce", False)
+            self.fp_scheme = self.config["dataset"].get("fp_scheme", "gaussian").lower()
+            self.fp_params = self.config["dataset"]["fp_params"]
+            self.save_fps = self.config["dataset"].get("save_fps", True)
+            self.cutoff_params = self.config["dataset"].get(
+                "cutoff_params", {"cutoff_func": "Cosine"}
+            )
+            descriptor_setup = (
+                self.fp_scheme,
+                self.fp_params,
+                self.cutoff_params,
+                self.elements,
+            )
+            self.train_dataset = AtomsDataset(
+                images=training_images,
+                descriptor_setup=descriptor_setup,
+                forcetraining=self.forcetraining,
+                # pca_reduce=self.pca_reduce,
+                # pca_setting=self.config["dataset"].get(
+                #     "pca_setting",
+                #     {"num_pc": 20, "elementwise": False, "normalize": False},
+                # ),
```

Comment:
Can  we remove these comments please. This can be added back at a later time for PCA customizability.